test/functional/terminal/api_spec.lua

```diff
@@ -66,5 +66,47 @@ describe('api', function()
 
     socket_session1:request("nvim_command", "qa!")
   end)
+
+  it("stopinsert RPC request during insert-mode", function()
+    -- Start the socket from the child nvim.
+    child_session.feed_data(":echo serverstart('"..socket_name.."')\n")
```

Comment:
`child_session.feed_data` is a hack for when we need to test a child Nvim (i.e. `nvim` running inside `:terminal`). I don't think that's needed for testing this change, using a regular session will avoid a lot of complexity.

I would move the test to `test/functional/terminal/edit_spec.lua` and probably the screen tests aren't needed, only need to check `nvim_get_mode`.

Reply:
I see, I should have spent more time trying to figure out what I was doing instead of blindly copy/pasting what I saw.

Thanks for the explanation :)